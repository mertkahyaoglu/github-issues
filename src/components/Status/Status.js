import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export default function Status({ status, icon, children, className }) {
  return (
    <StyledStatus status={status} className={className}>
      {icon && React.createElement(icon, { style: { marginRight: '0.5em' } })}
      {children}
    </StyledStatus>
  );
}

Status.defaultProps = {
  status: 'default',
};

Status.propTypes = {
  status: PropTypes.string.isRequired,
  icon: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  className: PropTypes.string,
};

const StyledStatus = styled.div`
  display: flex;
  align-items: center;
  color: ${props =>
    props.status === 'success' ||
    props.status === 'warning' ||
    props.status === 'default'
      ? props.theme[`${props.status}Color`]
      : props.theme.dangerColor};
`;
