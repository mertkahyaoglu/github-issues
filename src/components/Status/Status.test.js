import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Status from './Status';
import TestProviders from 'utils/TestProviders';
import Success from 'react-icons/lib/fa/check';
import Warning from 'react-icons/lib/fa/exclamation';
import Error from 'react-icons/lib/fa/bell';

describe('component/Status', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <TestProviders>
        <Status status="success" icon={Success}>
          Success
        </Status>
      </TestProviders>,
      div
    );
  });

  it('renders markup correctly', () => {
    const tree = renderer
      .create(
        <TestProviders>
          <Status status="success" icon={Success}>
            Success
          </Status>
        </TestProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders warning status correctly ', () => {
    const tree = renderer
      .create(
        <TestProviders>
          <Status status="warning" icon={Warning}>
            Warning
          </Status>
        </TestProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders error status correctly', () => {
    const tree = renderer
      .create(
        <TestProviders>
          <Status status="error" icon={Error}>
            Error
          </Status>
        </TestProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
