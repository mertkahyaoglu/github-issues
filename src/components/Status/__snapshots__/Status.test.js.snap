// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Status renders error status correctly 1`] = `
<div
  className="sc-bdVaJa cKnxhf"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "marginRight": "0.5em",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m20.4 37.9q0-0.4-0.4-0.4-1.3 0-2.3-1t-0.9-2.2q0-0.4-0.4-0.4t-0.3 0.4q0 1.6 1.1 2.8t2.8 1.1q0.4 0 0.4-0.3z m18.2-6.5q0 1.2-0.9 2t-2 0.9h-10q0 2.4-1.7 4t-4 1.7-4-1.7-1.7-4h-10q-1.2 0-2-0.9t-0.9-2q1.1-0.9 2.1-1.9t1.9-2.7 1.6-3.5 1.1-4.6 0.5-5.8q0-3.4 2.6-6.3t6.8-3.6q-0.1-0.4-0.1-0.9 0-0.8 0.6-1.5t1.5-0.6 1.5 0.6 0.6 1.5q0 0.5-0.1 0.9 4.2 0.6 6.8 3.6t2.6 6.3q0 3.1 0.5 5.8t1.1 4.6 1.6 3.5 1.9 2.7 2.1 1.9z"
      />
    </g>
  </svg>
  Error
</div>
`;

exports[`component/Status renders markup correctly 1`] = `
<div
  className="sc-bdVaJa hiZeDy"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "marginRight": "0.5em",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m37.3 12.6q0 0.9-0.6 1.6l-19.2 19.1q-0.6 0.7-1.5 0.7t-1.6-0.7l-11.1-11.1q-0.6-0.6-0.6-1.5t0.6-1.5l3.1-3q0.6-0.7 1.5-0.7t1.5 0.7l6.6 6.5 14.6-14.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.6 0.6 1.5z"
      />
    </g>
  </svg>
  Success
</div>
`;

exports[`component/Status renders warning status correctly  1`] = `
<div
  className="sc-bdVaJa xDNXI"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "marginRight": "0.5em",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m24.4 27.9v5q0 0.5-0.4 1t-1 0.4h-5.7q-0.6 0-1-0.4t-0.4-1v-5q0-0.6 0.4-1t1-0.5h5.7q0.6 0 1 0.5t0.4 1z m0.7-23.6l-0.6 17.1q0 0.6-0.5 1t-1 0.5h-5.7q-0.6 0-1-0.5t-0.5-1l-0.6-17.1q0-0.6 0.4-1t1-0.4h7.1q0.6 0 1 0.4t0.4 1z"
      />
    </g>
  </svg>
  Warning
</div>
`;
